### DISCUSSION TITLE
Cant enable JIT and WHDLoad crashes.

### DISCUSSION THREAD

**Discussion by TomBonk** [BlitterStudio/amiberry#1536] [General]
So I'm just playing around with my freshly purchased Raspberry Pi 5 and realized that I can't turn on JIT. The options are grayed out. Is this related to the Rpi 5 ?

I tested versions:

5.7.4
5.7.5
7.0.0-RC1

Second issue.
Many WHDLoad titles crash with a divide-by-zero error when the processor speed is set to “fastest”.

This error occurs in versions 5.7.4 and 5.7.5
In version 7.0.0-RC1 everything looks OK.

![1](https://github.com/user-attachments/assets/fd65b238-499d-41f5-9b6e-237aa3f074aa)
![2](https://github.com/user-attachments/assets/604bba61-f260-43f0-89dc-fd6366ad29da)

**Comment by midwan** [comment]
JIT for ARM is currently only available on Amiberry-Lite (v5).

**Reply by giantclambake** [reply]
@midwan 
I just noticed something looking at this ~ I doubt much can be done about it (maybe FAQ entry?)....

*If user has created a config.uae file associated with a whdload.lha game archive using a previous (standalone) version of amiberry, that config.uae file will fail if amiberry has been upgraded to a system-wide installation....the cause being the relocation of the `whdboot` directory....ie; from logfile...

```
my_getvolumeinfo: lstat on file /home/gcb/Amiberry/whdboot/boot-data.zip failed
directory '/home/gcb/Amiberry/whdboot/boot-data.zip' not found, mounting as empty drive.
my_getvolumeinfo: lstat on file /home/gcb/Amiberry/whdboot/save-data failed
directory '/home/gcb/Amiberry/whdboot/save-data' not found, mounting as empty drive.
```

Mentioning merely FTR, as it is/was a usage scenario that I'd not considered previously ;)

Addendum: It occurred to me, that if a user already has a previous version of amiberry installed (standalone guise), and then installs for example `amiberry_7.0.0_amd64.deb`, I believe it's the case that ~/Amiberry/whdboot/*  likely will still exist?

If so, user will have 2 whdboot directories, and if an 'old' config.uae file is sourced, it will use the 'old' whdboot directory.... and all newly launched/created  whdload.lha game archives and/or config.uae files, will use the 'new' whdboot directory (`~/.local/share/amiberry/whdboot`) which may/will contain updated files.... I doubt this situation is ideal =)

**Comment by giantclambake** [comment]
> Second issue.
> Many WHDLoad titles crash with a divide-by-zero error when the processor speed is set to “fastest”.

Could not recreate this using amiberry v5.7.4 on RPi4B/4G with (raspbian) Bookworm 64bit. KMSDRM or Xorg .... tried 10 or so whdload game archives from turran/retroplay, all working as expected..  ergo this is either specifically RPi5 related (I don't have that hardware), or, lack of details here about the issue could draw in other variables...

@TomBonk 

You say "many", but not "all" WHDLoad titles ~ can you provide a few filenames of the whdload.lha games archives that display this error?

You might also provide a logfile of this problem ~ the crash would appear to be coming from whdload itself, however it's good to check the autoloader and emulation startup are being processed correctly.

**Reply by giantclambake** [reply]
I overlooked the blindingly obvious =)....

![ex](https://imgur.com/KGLdoj8.png)

That, is not the amiberry whdbooter process ~ it appears to be some Workbench setup being emulated in amiberry, which itself has a whdload_usr installation ... which in turn is failing using amiberry-lite for the emulation ...(but not amiberry v7.0)...

...this would be why I couldn't fault the amiberry whdbooter ~ it's not involved here, and working as expected.

I'm slating this as being a 'red herring' ~ I very much doubt amiberry-lite/amiberry is participle here....ie; if the Workbench instance was a later AmiKit release, this behavior wouldn't surprise me using amiberry-lite, as AmiKit more or less requires a fully featured UAE engine...

HTH
